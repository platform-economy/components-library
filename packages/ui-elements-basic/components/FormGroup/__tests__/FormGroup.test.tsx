import * as React from 'react';
import * as TestRenderer from 'react-test-renderer';
import 'jest-styled-components';

import { FormGroup } from '../index';

describe('ui-elements-basic', () => {
  describe('Components', () => {
    describe('FormGroup', () => {
      it('should render with children horizontally', () => {
        const tree = TestRenderer
          .create(
            <FormGroup layout="horizontal">
              <input type="text" />
              <input type="button" />
            </FormGroup>,
          )
          .toJSON();
        expect(tree).toMatchSnapshot();
      });
      it('should render with children vertically', () => {
        const tree = TestRenderer
          .create(
            <FormGroup layout="vertical">
              <input type="text" />
              <input type="button" />
            </FormGroup>,
          )
          .toJSON();
        expect(tree).toMatchSnapshot();
      });
    });
  });
});
