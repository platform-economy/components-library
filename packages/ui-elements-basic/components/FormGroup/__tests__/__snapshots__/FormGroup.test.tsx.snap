// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ui-elements-basic Components FormGroup should render with children horizontally 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: start;
  -webkit-justify-content: flex-start;
  -ms-flex-pack: start;
  justify-content: flex-start;
  width: 100%;
}

<label
  className="c0"
>
  <input
    type="text"
  />
  <input
    type="button"
  />
</label>
`;

exports[`ui-elements-basic Components FormGroup should render with children vertically 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-items: flex-start;
  -webkit-box-align: flex-start;
  -ms-flex-align: flex-start;
  align-items: flex-start;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  width: 100%;
}

<label
  className="c0"
>
  <input
    type="text"
  />
  <input
    type="button"
  />
</label>
`;
